### PR TITLE
Fix Mac OS build process with SDL Framework. Add instructions for sound support with Mac OS build

### PR DESCRIPTION
**Pull request by remyroy**
The Mac OS build is broken with SDL Framework. Also, it does not support sound. This PR fix the SDL Framwork build on Mac OS and add instructions for sound support.

### CODE REVIEW

**Comment by remyroy** [comment]
This should be enough for SDL Framework. I'll put more attention in a separate PR to finish the Mac OS shared libraries build.

**Comment by kevingranade** [comment]
Ideally I'd like a confirmation from another mac user that this works, if
you are one please post whether this works on your system.
With the low number of mac users we have though, we can't afford to sit on
it forever, so if nobody shows up in a day or two we should go ahead and
merge.

**Comment by Treah** [comment]
Hey guys tested this out on my mac today below is the build details. It compiled and ran just fine with this pull request merged in, 

MacBook Air 13 
OSX Version: 10.10.4

**Comment by remyroy** [comment]
I will invest more time into making sure that sound works properly with the other OSX builds eventually.

File: src/cata_tiles.h [hunk 1]
```diff
@@ -6,8 +6,13 @@
 #include <wordexp.h>
 #endif
 
-#include "SDL2/SDL.h"
-#include "SDL2/SDL_ttf.h"
+#if (defined OSX_SDL_FW)
+#   include "SDL.h"
+#   include "SDL_ttf.h"
+#else
+#   include "SDL2/SDL.h"
```
Comment (kevingranade) [inline src/cata_tiles.h]: Instead of having these ifdefs scattered around, can you append "SDL2/" to the end of the paths you're setting in the Makefile?

**Comment by remyroy** [comment]
I've just talked with the SDL guys about this include path mess for cross-platform and OS X Framworks build.

They said we should be using `sdl2-config --cflags` and the various `pkg-config SDL2_image --cflags`, `pkg-config SDL2_ttf --cflags` and `pkg-config SDL2_mixer --cflags` for our cflags than use `#include <SDL.h>`, `#include <SDL_ttf.h>`, `#include <SDL_image.h>` and `#include <SDL_mixer.h>` instead of using those ifdefs.

Since `sdl2-config --cflags` (and the other pkg-config) should return something like:

```
-I/usr/include/SDL2 -D_REENTRANT
```

it should always include the SDL2 dir which mean we can avoid entering the path in our includes.

That would solve this problem without using a bunch of ifdefs. What do you think?

**Comment by kevingranade** [comment]
Yea I was hoping there was a solution like that, in general multiple path components in includes means you're doing something wrong.

**Comment by remyroy** [comment]
These last commits should fix things. #14072 depends on this PR.

I've tested this on OSX with Framework, OSX curses, Ubuntu native Tiles, Ubuntu native curses, cross-compiling with MXE to Windows Tiles, cross-compiling with MXE to Windows curses, MSYS2 Titles and MSYS2 curses and they all worked.

**Comment by remyroy** [comment]
I've just checked with @DanmakuDan which is a Code::Blocks user to see if this change would break his build process with the changes to the SDL includes. It seems like he does not rely on the provided project file `CataclysmWin.cbp` and he manually adds and remove include folders. Code::Blocks users will most likely have to manually include the SDL2 folder after this PR.

**Pull request merged**